src/main/java/soot/Scene.java

```diff
@@ -1869,6 +1869,10 @@ public void setPkgList(List<String> list) {
   /** Create an unresolved reference to a method. */
   public SootMethodRef makeMethodRef(SootClass declaringClass, String name, List<Type> parameterTypes, Type returnType,
       boolean isStatic) {
+    if (declaringClass.getName().equals(SootMethodRefHandleImpl.METHODHANDLE_SIGNATURE)
```

Comment:
The check may be put into a static method on SootMethodRefHandleImpl. E.g:
`PolymorphicIvokeMethodRef.handlesClass(declaringClass)`.

Ps: Better naming would be preferable...

Reply:
Thereby, if more classes with the SignaturePolymorphic annotation are added later on, there is a single point to add these..